### PR TITLE
Tweaks to respect namespace when using global.rbac.namespaced

### PR DESCRIPTION
**Pull request by jjcollinge**
Signed-off-by: Joni Collinge <jonathancollinge@live.com>

# Description

> NOTE: This is still WIP whilst I test various things. Also this is a hack aimed at getting a Dapr control plane deployment working within a namespace - this may bastardize things not originally intended for this use case so we can add new settings where required.

Once you enable `global.rbac.namespaced` the Dapr control plane gets RBAC permissions limited to a single namespace. Currently the Dapr control plane still tries to access resources outside of that namespace and then crashes.
Changes required to make this work:
- Update sentry to work the same as the operator and use the Kubernetes downward API to get the namespace it is deployed in and get the dapr configuration from there. (is this a breaking change?)
- Update the sidecar injector to do the same when loading the configuration. (is this a breaking change?)
- Update the mutating webhook to use a namespace selector limited to the deployment namespace.
- Add a new flag to the operator `no-crd-patch` which stops the operator trying to patch the cluster level CRD resources with the expectation that the CRDs will be installed and patched separately. This is needed as the operator will never have permission to access the cluster level CRD resources with the current RBAC restrictions. This needs further investigation to understand how to handle the cabundle in the conversion webhook properly externally.
- Updates the RBAC to NOT deploy cluster role or cluster role binding when running in `namespaced` mode and move permissions down to role. This needs further reviewing to see if it'll cause any issues.

## Issue reference

Please reference the issue this PR will close: https://github.com/dapr/dapr/issues/6611

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: charts/dapr/charts/dapr_rbac/templates/sentry.yaml [hunk 1]
```diff
@@ -61,6 +61,9 @@ rules:
   - apiGroups: ["dapr.io"]
     resources: ["configurations"]
     verbs: ["list"]
+  - apiGroups: ["authentication.k8s.io"]
+    resources: ["tokenreviews"]
+    verbs: ["create"]
```
Comment (JoshVanL) [inline charts/dapr/charts/dapr_rbac/templates/sentry.yaml:66]: Token reviews are not namespaced and must be cluster role bound.

**Review by JoshVanL — CHANGES_REQUESTED** [review]
How can we test a namespaced deployment of Dapr in CI?

File: charts/dapr/charts/dapr_rbac/templates/sentry.yaml (hunk 1 above)
Comment (jjcollinge) [reply charts/dapr/charts/dapr_rbac/templates/sentry.yaml:66]: Yes and I was waiting for this to cause me an issue with validation... not seen anything yet!

Comment (JoshVanL) [reply charts/dapr/charts/dapr_rbac/templates/sentry.yaml:66]: Yes, today this won't do much but will cause issues in 1.12 when clients begin requesting their certs from sentry for real.

**Comment by lguianu** [comment]
Hey @jjcollinge, have you been able to find a fix for when we enable global.rbac.namespaced and it's not working?

**Pull request closed without merging**